### ISSUE TITLE
Looping sized native typed array is slow

### ISSUE THREAD

**Issue by stphnlyd** [rakudo/rakudo#3532]
## The Problem

Comparing below two files, the only difference is that `@a` is sized in a.raku but not in b.raku. 

```
# cat a.raku
my int32 @a[1000000];
for (0 .. 999999) { @a[$_] = $_; }
```

```
# cat b.raku
my int @a;
for (0 .. 999999) { @a[$_] = $_; }
```

Their execution time differs a lot. Looping sized native typed array is much slower. 

```
$ time raku a.raku

real	0m3.514s
user	0m3.545s
sys	0m0.029s

$ time raku b.raku

real	0m0.179s
user	0m0.204s
sys	0m0.016s
```

## Environment
* Operating system: Ubuntu 18.04 LTS
* Compiler version (`perl6 -v`): This is Rakudo version 2020.01 built on MoarVM version 2020.01.1
implementing Perl 6.d.

**Comment by lizmat** [comment]
Some more datapoints:

    $ raku -e 'my int32 @a[1000000]; @a[$_] = $_ for ^1000000'
    real	0m3.365s
    user	0m3.391s
    sys	0m0.027s

    $ raku -e 'my int32 @a[1000000]; @a.WHAT.^compose; @a[$_] = $_ for ^1000000'
    real	0m0.214s
    user	0m0.244s
    sys	0m0.029s

So it looks like it's a simple matter of making sure we call `compose` on its meta-object?

**Comment by lizmat** [comment]
Fixed with https://github.com/rakudo/rakudo/commit/80f2aebff4  .  Not sure how we can test this reliably.  Suggestions welcome.

**Comment by Altai-man** [comment]
A nice patch!
AFAIK, the policy for performance tests is "don't write them" as they create more issues than solve problems.

**Comment by lizmat** [comment]
Had to revert https://github.com/rakudo/rakudo/commit/80f2aebff4

**Comment by lizmat** [comment]
I think this will also be affected by @jnthn's current work, so assigning this ticket to him

**Comment by MasterDuke17** [comment]
Fixed with #4084
